src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -154,46 +154,40 @@ void RegisterAppInterfaceRequest::Run() {
   // FIXME(EZamakhov): on shutdown - get freez
 
   // wait till HMI started
-  while (ApplicationManagerImpl::exists() &&
-         !ApplicationManagerImpl::instance()->IsStopping() &&
-         !ApplicationManagerImpl::instance()->IsHMICooperating()) {
+  while (!application_manager_.IsStopping() &&
+         !application_manager_.IsHMICooperating()) {
     LOG4CXX_DEBUG(logger_,
-                  "Waiting for the HMI... conn_key=" << connection_key()
-                                                     << ", correlation_id="
-                                                     << correlation_id()
-                                                     << ", default_timeout="
-                                                     << default_timeout()
-                                                     << ", thread="
-                                                     << pthread_self());
-    ApplicationManagerImpl::instance()->updateRequestTimeout(
+                  "Waiting for the HMI... conn_key="
+                      << connection_key()
+                      << ", correlation_id=" << correlation_id()
+                      << ", default_timeout=" << default_timeout()
+                      << ", thread=" << pthread_self());
+    application_manager_.updateRequestTimeout(
         connection_key(), correlation_id(), default_timeout());
     sleep(1);
     // TODO(DK): timer_->StartWait(1);
   }
 
-  if (!ApplicationManagerImpl::exists()) {
```

Comment:
@VProdanov , yes. Because AM::singleton was removed. And every command have reference to AM(and we don`t need to check it).